### DISCUSSION TITLE
How to actually add new jobs over network?

### DISCUSSION THREAD

**Discussion by wirelessfuture** [go-co-op/gocron#379] [General]
Let's say I create a scheduler instance and have it running in its own docker container in blocking mode. Is there a way I can add new jobs to the scheduler over the network? 

In order to do this, the process adding new jobs needs to have access to the same scheduler instance as running in Docker container right? 

Does anyone have some ideas on how to make this work?

**Comment by JohnRoesler** [comment]
> In order to do this, the process adding new jobs needs to have access to the same scheduler instance as running in Docker container right?

Yes

> Does anyone have some ideas on how to make this work?

I would suggest building an API that accepts a representation of a job and adds it to the scheduler when called. You'll probably want to start the scheduler Async instead of blocking in this case so that you can have a router/mux and an http or gRPC listener.

**Reply by wirelessfuture** [reply]
Hey John, thanks for commenting. Do you know of any examples in the wild that shows this in action?

**Reply by JohnRoesler** [reply]
I am not aware of any. It's possible / likely people are doing that but I haven't seen it.
